### ISSUE TITLE
ruff analyze graph omits excluded paths, so modules only imported from them appear unused

### ISSUE THREAD

**Issue by markjm** [astral-sh/ruff#23255]
### Question

I am not sure if this counts as a bug because it may be considered as expected behaviour, but reporting anyway in case folks agree with my findings

Repro available in https://gist.github.com/markjm/b7077d64c126e63cae8c9efca97a5877

----

## Actual
```bash
$ ruff analyze graph .
warning: `ruff analyze graph` is experimental and may change without warning
{
  "pkg/__init__.py": []
}

$ ruff analyze graph ./excluded/__init__.py
warning: `ruff analyze graph` is experimental and may change without warning
{
  "excluded/__init__.py": [
    "pkg/__init__.py"
  ]
}
```
## Expected
Files excluded from other ruff commands are included in analyze

### Discussion

From what I understand, this is happening because ruff's excludes intended for its other tools (lint/formatting) are also being applied to analyze. While it's not neccesarily wrong, I think it does produce unexpected results. I want to exclude stuff from static analysis and formatting for various reasons (codegen, etc), but still would like to see it in my project graph. It seems I could use something like `--isolated` to not read my configs at all, but there are other settings which analyze does want to use (`namespace-packages`, `src`), so not sure that is best.

Would it make sense to not respect `excludes` configs for `analyze` or seperate them out to a seperate `[tool.ruff.analyze]` config?

Thanks!

### Version

ruff 0.15.0

**Comment by ntBre** [comment]
If I understand correctly, I think the more granular settings you want already exist. There's:
- the top-level [exclude](https://docs.astral.sh/ruff/settings/#exclude) setting that applies to all sub-commands
- [analyze.exclude](https://docs.astral.sh/ruff/settings/#analyze_exclude)
- [format.exclude](https://docs.astral.sh/ruff/settings/#format_exclude)
- [lint.exclude](https://docs.astral.sh/ruff/settings/#lint_exclude)

So it sounds like you need to move some of your top-level exclusions into `lint.exclude` and/or `format.exclude`. You may also be able to use a negated glob like `!/path/to/include` in `analyze.exclude` to override the global exclusions, but I'd have to double check our globbing code.

**Comment by markjm** [comment]
Oh interesting, let me play around with the globs. I am currently using `extend-excludes` which looks to only be available at the top level, which may have caused some of my confusion!
